### ISSUE TITLE
"Feature relevance quantification in explainable AI: A causal problem" by Janzing et al.

### ISSUE THREAD

**Issue by mlds2020** [NorskRegnesentral/shapr#215]
Doesn't the recent paper by [Janzing et al. (2020)](http://proceedings.mlr.press/v108/janzing20a/janzing20a.pdf) invalidate the theoretical justification behind why SHAP is using its "simplified function" (while the function used is indeed correct) - thus invalidating any consequent work improving on this false cornerstone (i.e. the assumed independence of features) - which includes **SHAPR**?

Would you suggest we should rather keep using SHAP?

**Comment by mlds2020** [comment]
[Sundararajan and Najmi (2019)](https://arxiv.org/pdf/1908.08474.pdf) point out this flaw too, for that matter.

**Comment by nikolase90** [comment]
CC: @martinju

**Comment by martinju** [comment]
Hi, and sorry for the late reply, its holiday time Norway these days. 

I totally understand your confusion. We am perfectly aware of these papers, but are still confident that using the conditional expectation as the contribution function gives the most appropriate explanation. 
Not accounting for the dependence between the features requires evaluation of your model at potentially impossible feature combinations which the model has no reason to provide reasonable predictions for. 

From our point of view, the reason the two papers are reaching different conclusions, is that they they base their work on an incorrect fundament, essentially saying that all information is given by the model itself.

Consider for instance example 1 in Janzig et al, where f(x1,x2)=x1. The authors state without any discussion that "Obviously feature x2 is irrelevant". I completely disagree about this claim, particularly since they also say that x1=x2 with probability 1, so if you have only observed x2, you also know the value of x1 (and the interpretation of shapley value explanations are indeed of this missing value type). So how can they say that x2 is irrelevant? Seems rather odd to me.

Since we disagree on the fundamentals, which their simulations and conclusions are based on, I don't find them relevant.

Bottom line: I would still very much recommend using our approach and explain models using our R-package.

While I have not had time to read it yet, you may find the following recent paper relevant for this topic: https://arxiv.org/pdf/2006.16234.pdf

I hope this answers your questions. Feel free to re-open the issue if you have further questions 🙂
